hooks/open-telemetry/pom.xml

```diff
@@ -28,9 +28,13 @@
 		<!-- we inherent dev.openfeature.javasdk and the test dependencies from the parent pom -->
 		<dependency>
 			<groupId>io.opentelemetry</groupId>
-			<artifactId>opentelemetry-sdk</artifactId>
+			<artifactId>opentelemetry-api</artifactId>
```

Comment:
Do we need to specify a minimum version now? Particularly because of metrics?

Reply:
restructured the pom to reduce confusion